### PR TITLE
fix(venv): use shell agnostic script to generate env files and support

### PR DESCRIPTION
**Pull request by fffonion**
docker compose plugin

### Summary

This reduces the burden of developer needs to learn fish/zsh/bash syntax to add features to the venv. Now only the skleton is depending on each shell's flavor, the actual logic to set env vars and start docker-compose is always in bash.

### Checklist

- [x] The Pull Request has tests
- [na] There's an entry in the CHANGELOG
- [na] There is a user-facing docs PR against https://github.com/Kong/docs.konghq.com - PUT DOCS PR HERE

### Full changelog

* [Implement ...]

### Issue reference

Fix KAG-441, KAG-1350

### CODE REVIEW

File: scripts/dependency_services/up.sh [hunk 1]
```diff
@@ -1,43 +1,38 @@
...
     echo "You must source this script: \$ source $0" >&2
     exit 33
 fi
 
+export KONG_SERVICE_ENV_FILE=$(mktemp)
+
 if [ -n "$ZSH_VERSION" ]; then
     cwd=$(realpath $(dirname $(readlink -f ${(%):-%N})))
 else
     cwd=$(realpath $(dirname $(readlink -f ${BASH_SOURCE[0]})))
```
Comment (outsinre) [inline scripts/dependency_services/up.sh:13]: `realpath` and `readlink` is duplicate.

**Comment by outsinre** [comment]
LGTM except a few suggestions.

**Review by outsinre — APPROVED** [review]

**Review by attenuation — APPROVED** [review]
LGTM， But I have some concerns about venv we don't have any task manager tools, For example, we started a venv but suddenly I lose my SSH connection. How do we manage the old development environment? We must give the user an interface to clean all of the old venv docker-compose instances.

**Comment by fffonion** [comment]
@attenuation we can go into the venv again then `start_services` (which will be a nop) and then `stop_services`, i tend to not expose `stop_services` at the beginning to make the state machine more concise. If this becomes a concern we can revisit this :)

**Pull request merged**